ios/AppLovinMAXNativeAdView.m

```diff
@@ -0,0 +1,258 @@
+// properties
+@property (nonatomic, copy) NSString *adUnitIdentifier;
+@property (nonatomic, copy) NSString *placement;
+@property (nonatomic, copy) NSString *customData;
+@property (nonatomic, copy) NSDictionary *extraParameters;
+@property (nonatomic, copy) RCTDirectEventBlock onNativeAdLoaded;
+
+@end
+
+@implementation AppLovinMAXNativeAdView
+
+static NSString *NativeAdLoaderErrorAdUnitIdentifier;
+
++ (NSString *)errorAdUnitIdentifier
+{
+    return NativeAdLoaderErrorAdUnitIdentifier;
+}
+
+- (instancetype)initWithBridge:(RCTBridge *)bridge
+{
+    if ( self = [super init] )
+    {
+        RCTLogInfo( @"AppLovinMAX: NativeAdView: init: %p", self );
+        
+        self.bridge = bridge;
+        self.loadingAd = NO;
+    }
+    return self;
+}
+
+- (void)dealloc
+{
+    RCTLogInfo( @"AppLovinMAX: NativeAdView: dealloc: %p", self );
+    
+    [self destroyAd];
+}
+
+- (void)setAdUnitId:(NSString *)adUnitId
+{
+    if ( self.loadingAd ) return;
+    
+    RCTLogInfo( @"AppLovinMAX: NativeAdView: setAdUnitId: %@", adUnitId );
+    
+    self.adUnitIdentifier = adUnitId;
+    self.loadingAd = YES;
+    [self loadAd];
+}
+
+- (void)setMediaView:(NSNumber *)tag
+{
+    if ( !self.nativeAd ) return;
+    
+    RCTLogInfo( @"AppLovinMAX: NativeAdView: %p setMediaView: %@ for %@", self, self.adUnitIdentifier, tag );
+    
+    [self addStretchView: self.nativeAd.nativeAd.mediaView toItem: [self getNativeView: tag]];
+}
+
+- (void)setOptionsView:(NSNumber *)tag
+{
+    if ( !self.nativeAd ) return;
+    
+    RCTLogInfo( @"AppLovinMAX: NativeAdView: setOptionsView: %@", self.adUnitIdentifier );
+    
+    [self addStretchView: self.nativeAd.nativeAd.optionsView toItem: [self getNativeView: tag]];
+}
+
+- (void)setIconImage:(NSNumber *)tag
+{
+    if ( !self.nativeAd ) return;
+    
+    RCTLogInfo( @"AppLovinMAX: NativeAdView: setIconImage: %@", self.adUnitIdentifier );
+    
+    UIView *view = [self getNativeView: tag];
+    if ([view isKindOfClass:[UIImageView class]])
+    {
+        [self setImageView: self.nativeAd.nativeAd.icon.image toView: (UIImageView *) view];
+    }
+}
+
+- (void)loadAd
+{
+    RCTLogInfo( @"AppLovinMAX: NativeAdView: %p loadAd: %@", self, self.adUnitIdentifier );
+    
+    [self load: self.adUnitIdentifier : self.placement : self.customData : self.extraParameters];
+}
+
+- (void)destroyAd
+{
+    if ( self.nativeAd )
+    {
+        RCTLogInfo( @"AppLovinMAX: NativeAdView: destroyCurrentAd: %p", self.nativeAd );
+        
+        if (self.nativeAd.nativeAd)
+        {
+            [self.nativeAd.nativeAd.mediaView removeFromSuperview];
+            [self.nativeAd.nativeAd.optionsView removeFromSuperview];
+        }
+        
+        [self.nativeAdLoader destroyAd: self.nativeAd];
+    }
+    
+    if ( self.nativeAdView )
+    {
+        [self.nativeAdView removeFromSuperview];
+    }
+}
+
+- (void)load:(NSString *)adUnitIdentifier :(NSString *)placement :(NSString *)customData :(NSDictionary *)extraParameters
```

Comment:
How come no need for `RCT_EXPORT_METHOD` ? Would be nice to have some sort of indicator that this can be called form react layer.

Reply:
RCT_EXPORT_METHOD is defined on the manager side.   On the view side, the function is called from the manager side depending on the view tag (ref).   So, this should be okay.

Reply:
If it is not declared for React Native layer to invoke, we should name the method signature properly with argument labels